internal/control_plane/server/execution/execution.go

```diff
@@ -2,35 +2,117 @@ package execution
+	executorRegistered(ctx context.Context, id string) (bool, error)
+	UpdateExecutorStatus(ctx context.Context, request *executorCPproto.Ping) (*executorCPproto.HealthResponse, error)
 }
 
 type execution struct {
-	metadata repository.MetadataRepository
+	metadataRepo      metadataRepo.MetadataRepository
+	executorRepo      executorRepo.ExecutorRepository
+	activeExecutorMap *sync.Map
 }
 
 // NewExec creates a new instance of metadata respository
-func NewExec(metadataRepo repository.MetadataRepository) Execution {
+func NewExec(metadataRepo metadataRepo.MetadataRepository, executorRepo executorRepo.ExecutorRepository) Execution {
 	return &execution{
-		metadata: metadataRepo,
+		metadataRepo:      metadataRepo,
+		executorRepo:      executorRepo,
+		activeExecutorMap: new(sync.Map),
 	}
 }
 
-//SaveMetadataToDb calls the metadata/repository Save() function and returns MetadataName
-func (e *execution) SaveMetadataToDb(ctx context.Context, metadata *protobuf.Metadata) (*protobuf.MetadataName, error) {
-	result, err := e.metadata.Save(ctx, metadata.Name, metadata)
+//SaveMetadataToDb calls the repository/metadata Save() function and returns MetadataName
+func (e *execution) SaveMetadataToDb(ctx context.Context, metadata *clientCPproto.Metadata) (*clientCPproto.MetadataName, error) {
+	result, err := e.metadataRepo.Save(ctx, metadata.Name, metadata)
 	return result, err
 }
 
-//ReadAllMetadata calls the metadata/repository GetAll() and returns MetadataArray
-func (e *execution) ReadAllMetadata(ctx context.Context) (*protobuf.MetadataArray, error) {
-	result, err := e.metadata.GetAll(ctx)
+//ReadAllMetadata calls the repository/metadata GetAll() and returns MetadataArray
+func (e *execution) ReadAllMetadata(ctx context.Context) (*clientCPproto.MetadataArray, error) {
+	result, err := e.metadataRepo.GetAll(ctx)
 	return result, err
 }
+
+func (e *execution) RegisterExecutor(ctx context.Context, request *executorCPproto.RegisterRequest) (*executorCPproto.RegisterResponse, error) {
+	key := request.ID
+	res, err := e.executorRepo.Save(ctx, key, request)
+	return res, err
+}
+
+func (e *execution) StartHealthCheck(ctx context.Context, activeExecutorMap *sync.Map, id string, healthChan chan string) {
+	ticker := time.NewTicker(time.Second)
+	status := make(chan bool)
+	deadline := 200
+	presentTime := 0
+
+	for {
+		select {
+		case health := <-healthChan:
+			err := e.executorRepo.UpdateExecutorStatus(ctx, id, health)
+			if err != nil {
+				logger.Error(errors.New(fmt.Sprintf("error in updating status for executor with %s id", id)), "")
+				status <- true
+			}
+			presentTime = 0
+			logger.Info(fmt.Sprintf("health status for executor with %s id is %s", id, health))
+		case expired := <-status:
+			if expired {
+				err := e.executorRepo.UpdateExecutorStatus(ctx, id, "expired")
+				if err != nil {
+					logger.Error(errors.New(fmt.Sprintf("error in updating status for executor with %s id", id)), "")
```

Comment:
put the err returned from e.executorRepo.UpdateExecutorStatus(ctx, id, "expired") to the logger.